docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -231,23 +240,31 @@ type ClusterClassSpec struct {
   // Note: Patches will be applied in the order of the array.
   // +optional
   Patches []ClusterClassPatch `json:"patches,omitempty"`
+  
+  // MachineHealthCheckClass defines a MachineHealthCheck for each 
+  // ControlPlaneClass and MachineDeploymentClass in this ClusterClass.
+  // +optional
+  MachineHealthCheck  *MachineHealthCheckClass `json:"machineHealthCheck,omitempty"`
```

Comment:
I think we should be specific here and keep it as is. 

I think we can always add other features with additional individual fields and if we want to group them at some point we can deprecate + merge.

For now it would be very speculative as we don't know how/if other features are added. 
* ClusterResourceSet seems to be out-of-scope of ClusterClass at the moment (as it's orthogonal to ClusterClass and applies across multiple clusters)
* Autoscaler: 
    * labels can be already added today with our metadata fields
    * deploying the autoscaler itself seems to rather fit into ClusterAddons bucket (https://github.com/kubernetes-sigs/cluster-api/issues/5491)

Overall ClusterClass itself is all about the cluster lifecycle, so I'm not sure an additional field/level like `integrations` or `lifecycleIntegrations` would add a lot of benefit and depending on which and how we add additional features the field name might not fit.

@elmiko WDYT?

Reply:
@sbueringer i think your reasoning makes good sense. i would be happy to see the integration defined better in an enhancement that comes out of #5491.

Reply:
Could we move this definition under `WorkersClass`? And specify that this one will be applied to each one of the MachineDeployment classes unless they supply their own configuration/overrides?

Reply:
I think the most straightforward implementation is to define this on the lowest levels. If we want overriding behaviour later on we can introduce the higher level workerClass MHC.

Reply:
Having the MHC Class top level can be confusing to users, because it's one level above ControlPlane so folks might think that there is some layering in place.

Reply:
+1 to have MHC only inside CP and MD for the first iteration (and thus to remove it from here)